nemo/collections/asr/data/huggingface/hf_audio_to_text.py

```diff
@@ -0,0 +1,592 @@
+            if lang is not None:
+                text_tokens = self.parser(text, lang)
+            # for future use if want to add language bypass to audio_to_text classes
+            # elif hasattr(parser, "lang") and parser.lang is not None:
+            #    text_tokens = parser(text, parser.lang)
+            else:
+                raise ValueError("lang required in manifest when using aggregate tokenizers")
+        else:
+            text_tokens = self.parser(text)
+        text_tokens_length = len(text_tokens)
+        if self.bos_id is not None:
+            text_tokens = [self.bos_id] + text_tokens
+            text_tokens_length += 1
+        if self.eos_id is not None:
+            text_tokens = text_tokens + [self.eos_id]
+            text_tokens_length += 1
+        return text_tokens, text_tokens_length
+
+
+def get_nested_dict_value(dictionary: dict, key: str):
+    """
+    the key should be a string of nested keys separated by `.`, e.g. `key1.key2.key3`,
+    then the returned value will be `dictionary[key1][key2][key3]`
+    """
+    nested_keys = key.split(".")
+    result = dictionary
+    for k in nested_keys:
+        if k not in result:
+            raise KeyError(
+                f"Key `{key}` not found in [{result.keys()}], target is {nested_keys}, input is {dictionary}"
+            )
+        result = result[k]
+    return result
+
+
+class _HFAudioTextDataset(Dataset):
+    def __init__(
+        self,
+        audio_key: str,
+        text_key: str,
+        sample_rate_key: str,
+        hf_data_cfg: DictConfig,
+        parser: Union[str, Callable],
+        sample_rate: int,
+        augmentor: 'nemo.collections.asr.parts.perturb.AudioAugmentor' = None,
+        trim: bool = False,
+        bos_id: Optional[int] = None,
+        eos_id: Optional[int] = None,
+        pad_id: int = 0,
+        return_sample_id: bool = False,
+        channel_selector: Optional[ChannelSelectorType] = None,
+        normalize_db: Optional[float] = None,
+        ref_channel: Optional[int] = None,
+        id_key: Optional[str] = None,
+        normalize_text: bool = False,
+        symbols_to_keep: Optional[str] = None,
+    ) -> None:
+        super().__init__()
+        self.audio_key = audio_key
+        self.text_key = text_key
+        self.sample_rate_key = sample_rate_key
+        self.id_key = id_key
+        self.sample_rate = sample_rate
+        self.augmentor = augmentor if augmentor is not None else AudioAugmentor()
+        self.trim = trim
+        self.return_sample_id = return_sample_id
+        self.channel_selector = channel_selector
+        self.normalize_db = normalize_db
+        self.ref_channel = ref_channel
+
+        self.text_processor = HFTextProcessor(parser, bos_id, eos_id, pad_id, normalize_text, symbols_to_keep)
+
+        with open_dict(hf_data_cfg):
+            # streaming must be False for random access dataset
```

Comment:
Note / warn of streaming was set to true here and you're forcing it to false

Reply:
done